node/cn/tracers/api.go

```diff
@@ -891,7 +891,18 @@ func (api *CommonAPI) TraceCall(ctx context.Context, args kaiaapi.CallArgs, bloc
 	if config != nil && config.Reexec != nil {
 		reexec = *config.Reexec
 	}
-	statedb, release, err := api.backend.StateAtBlock(ctx, block, reexec, nil, true, false)
+
+	var (
+		statedb *state.StateDB
+		release StateReleaseFunc
+	)
+	// if next block does exist, use `StateAtTransaction` with transaction index zero. Otherwise, fallback to use `StateAtBlock` with the input block number
+	if next, err := api.blockByNumber(ctx, rpc.BlockNumber(block.NumberU64())); err != nil {
```

Comment:
What is this `next`? isn't it the same as `block` local variable? The geth branch (https://github.com/ethereum/go-ethereum/blob/da17f2d65b8922a20f86740644b2d7f6ef688e9b/eth/tracers/api.go#L948-L952) is about TxIndex and we don't take TxIndex, so in our code single StateAtTransaction(block, 0, ... ) should suffice.

Reply:
Yes `StateAtTransaction(101, 0, ...) equals to StateAtBlock(100, ...)+Initialize(101)`.

Oh, to preserve the semantics, we should do either `StateAtBlock(num)` or `StateAtTransaction(num+1)`...
In that case, can you revert to `StateAtBlock(num)`? Sorry I was confused.